sequential/insert_element_at/src/insert_element_at.erl

```diff
@@ -2,5 +2,15 @@
 
 -export([insert/3]).
 
-insert(_List, _Pos, _Element) ->
-  believe_in_yourself.
+insert(List, Pos, Element) ->
+    insert(List, Pos, Element, 1, []).
+
+insert([], _Pos, _Element, _Index, _Acc) -> [];
+insert([#{current := Current, old := _Old} | Tail], Pos,
```

Comment:
This code looks overly complex, unless I'm missing something you should be able to do this without a map and `++`. (a single list:reverse is probably ok)

Reply:
sorry I didn't see the description of the problem, disregard the part about the map